### PR TITLE
fix bot counterbattery NPE

### PR DESCRIPTION
**Pull request by NickAragua**
Fixes an observed NPE when the bot was targeting off-board units for counterbattery fire with on-board artillery. We do this by introducing a number of null checks, and short-circuiting some computations.

Also improves exception logging, as it was only logging the exception and not even the line number.

### CODE REVIEW

File: megamek/src/megamek/client/bot/BotClient.java [hunk 1]
```diff
@@ -588,7 +588,14 @@ private synchronized boolean calculateMyTurnWorker() {
...
         } catch (Throwable t) {
-            MegaMek.getLogger().error(t.toString());
+            StringBuilder sb = new StringBuilder();
+            sb.append(t.toString()).append("\r\n");
+           
+            for (StackTraceElement traceElement : t.getStackTrace()) {
+                sb.append(traceElement.toString());
+                
+            }
+            MegaMek.getLogger().error(sb.toString());
```
Comment (sixlettervariables) [inline megamek/src/megamek/client/bot/BotClient.java:598]: Can we not just pass `t` to error?

**Review by sixlettervariables — APPROVED** [review]
One comment, but otherwise approved.

**Comment by sixlettervariables** [comment]
It would be nice if instead of `null` we had an "offboard" hex we could use.

File: megamek/src/megamek/client/bot/BotClient.java (hunk 1 above)
Comment (NickAragua) [reply megamek/src/megamek/client/bot/BotClient.java:598]: Yep. Much easier.

**Comment by NickAragua** [comment]
Yeah, although I'm reluctant to make that kind of change, as there's probably a good amount of logic that depends on the offboard hex being null when it's offboard.

**Pull request merged**